src/menelaus/ensemble.py

```diff
@@ -23,10 +23,117 @@ def eval_simple_majority(detectors):
         return None
 
 
-# Keeping all evaluator functions in a table enables
-# ensembles to look up via str value, rather than
-# needing a function passed in.
-evaluators = {"simple-majority": eval_simple_majority}
+def eval_minimum_approval(approvals_needed=1):
+    """
+    Evaluator function that determines drift based on whether
+    a minimum number of provided detectors have alarmed. This
+    threshold can be 1 to the maximum number of detectors.
+
+    Note that this function is a closure, and so is called a
+    little differently:
+
+    Args:
+        approvals_needed (int): Minimum number of detectors that
+            must alarm for the ensemble to alarm.
+
+    Returns:
+        function (list -> str): Function that takes a list
+            of detectors and returns drift state.
+    """
+
+    def f(detectors):
+        """
+        Function that actually operates according to the minimum
+        approval scheme.
+
+        Args:
+            detectors (list): Detectors to use for alarming.
+
+        Returns:
+            str: ``"drift_state"`` if drift is determined, or ``None``
+        """
+        num_approvals = 0
+        for det in detectors:
+            if det.drift_state == "drift":
+                num_approvals += 1
+            if num_approvals >= approvals_needed:
+                return "drift"
+        return None
+
+    return f
+
+
+def eval_confirmed_approval(approvals_needed=1, confirmations_needed=1):
+    """
+    Evaluator that determines drift based on whether:
+        1) An initial ``a`` count of detectors alarmed for drift.
+        2) A subsequent ``c`` count of detectors confirmed drift.
+
+    Hypothethically, the distinction between this and
+    `eval_minimum_approval(a+c)`, is if the detectors were added to
+    a collection in a meaningful order. As such this voting
+    scheme iterates over detectors in preserved order of insertion into
+    the user-defined list, and uses the first ``approvals_needed``
+    amount for initial detection, and the next ``confirmations_needed``
+    amount for confirmation of drift.
+
+    Note that this function is a closure, and so is called a
+    little differently:
+
+    Args:
+        approvals_needed (int): Minimum number of detectors that
+            must alarm for the ensemble to alarm.
+        confirmations_needed (int): Minimum number of confirmations
+            needed to alarm, after `approvals_needed` alarms have been
+            observed.
+
+    Returns:
+        function (list -> str): Function that takes a list of detectors
+            and returns drift state.
+    """
+
+    def f(detectors):
```

Comment:
@tms-bananaquit Oh I see, you're getting at a sort of 'tracking' of the drift state (rather than a single point of voting at an update call), so that we may begin to account for waiting periods. 

This code definitely does not do that -- but unless you think otherwise I'll put that in #100